src/http/request_handler.rs

```diff
@@ -60,7 +60,10 @@ pub async fn graphql_request<T: DeserializeOwned + GraphQLRequestLike>(
   let request = serde_json::from_slice::<T>(&bytes);
   match request {
     Ok(request) => {
-      let mut response = request.data(req_ctx.clone()).execute(&server_ctx.schema).await;
+      let mut response = request
+        .data(req_ctx.clone())
+        .execute(&*server_ctx.schema.read().await)
```

Comment:
Reading the schema asynchronously isn't a great idea. `read` should be sync. It should be updated asynchronously.

Reply:
> Reading the schema asynchronously isn't a great idea. `read` should be sync. It should be updated asynchronously.

@tusharmath `tokio::sync::RwLock` is better in our case as `std::sync::RwLock` do not have any order guaranteed, the decision to provide read or write lock depends upon the OS, but in case of `tokio::sync::RwLock` if there is a request for write lock on at the head of the queue, it will not let out read locks 

referance: [std::sync::RwLock](https://doc.rust-lang.org/std/sync/struct.RwLock.html) and [tokio::sync::RwLock](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html)

Reply:
@tusharmath but `std::sync::RwLockReadGuard` do not have `Send` trait, hence can't be used with threads anyways

Reply:
do we have clear solution here?

we anyway need to some kind of locking to update the config from another task, right? So it will introduce some overhead anyway

Reply:
Yes, I tried with AtomicPtr, bit heap memory is slow and I can't find a way to get it perfectly synced. I think RwLock is the only solution I know as of now.